packages/truffle-compile/test/compilerSupplier.js

```diff
@@ -0,0 +1,189 @@
+const assert = require("assert");
+const sinon = require("sinon");
+const fs = require("fs");
+const CompilerSupplier = require("../compilerSupplier");
+let expectedResult, expectedFileName, supplier, config;
+
+describe("CompilerSupplier", () => {
+  describe("prototype.load()", () => {
+    describe("when a docker tag is specified in the config", () => {
+      beforeEach(() => {
+        config = {
+          docker: "randoDockerTagzzz",
+          version: "0.4.25"
+        };
+        supplier = new CompilerSupplier(config);
+        sinon.stub(supplier, "getBuilt");
+      });
+      afterEach(() => {
+        supplier.getBuilt.restore();
+      });
+
+      it("calls getBuilt with 'docker' as an argument", (done) => {
+        supplier.load()
+          .then(() => {
+            assert(supplier.getBuilt.calledWith("docker"));
+            done();
+          });
+      });
+    });
+
+    describe(`when the version is set to a specific word I can't write here`, () => {
+      // Because of how the test script is set up, you can't write native in the strings above and below
```

Comment:
Might be worth a `HACK` comment so it's searchable later

Reply:
Sounds good